cardano-api/src/Cardano/Api/IO.hs

```diff
@@ -117,18 +123,39 @@ handleFileForWritingWithOwnerPermission path f = do
     (targetDir, targetFile) = splitFileName path
 #endif
 
+readByteStringFile :: ()
```

Comment:
Tbh, a typeclass would work here better instead of having all those specialized functions.

Reply:
I don't have an objection to type classes.

You're welcome to put up a PR.